rcl/test/rcl/test_events.cpp

```diff
@@ -730,6 +733,35 @@ TEST_F(TestEventFixture, test_bad_get_handle)
   EXPECT_EQ(NULL, rcl_event_get_rmw_handle(NULL));
 }
 
+/*
+ * Test cases for the event_is_valid function
+ */
+TEST_F(TestEventFixture, test_event_is_valid)
+{
+  EXPECT_FALSE(rcl_event_is_valid(nullptr));
+  rcl_reset_error();
```

Comment:
```suggestion
  EXPECT_TRUE(rcl_error_is_set());
  rcl_reset_error();
```

By checking [rcl_error_is_set](https://github.com/ros2/rcl/blob/966efa3c49258cd011bbe2e78ebd718308bcb615/rcl/include/rcl/error_handling.h#L41), you make sure the implementation is actually setting an error message.

This applies everywhere else before `rcl_reset_error()`.